File: baekjoon/Python/2839_설탕_배달.py
N = int(input())
# 최소개수 => 5킬로 봉투를 최대한 사용, 3킬로만 사용할 수도 있으니
# N//5 ~ 0까지 루프
for cnt_5 in range(N // 5, -1, -1):
    cnt_3 = (N - 5 * cnt_5) // 3  # 5킬로 봉투량 뺀 무게를 3킬로 봉지수로 계산

    if cnt_5 * 5 + cnt_3 * 3 == N:
        print(cnt_5 + cnt_3)
        break
else:
    print(-1)

File: baekjoon/Java/2839_설탕_배달.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        int result = -1;

        // 5kg 봉지 최대개수부터 0까지 역순조회
        for (int cnt_5 = N / 5; cnt_5 >= 0; cnt_5--) {
            // 남은 무게 3kg 봉지
            int cnt_3 = (N - (cnt_5 * 5)) / 3;
            if (cnt_5 * 5 + cnt_3 * 3 == N) {
                result = cnt_5 + cnt_3;
                break;  // 조건이 참일 때만 루프 종료
            }
        }
        System.out.println(result);
    }
}
